cmd/kubeadm/app/util/tokens_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+func TestUsingEmptyTokenFails(t *testing.T) {
+	// Simulates what happens when you omit --token on the CLI
+	s := newSecretsWithToken("")
+
+	ok, err := UseGivenTokenIfValid(s)
+	assert.False(t, ok)
+	assert.NoError(t, err)
```

Comment:
Generally you want to check for errors first. If an error occurred, what does `ok` even mean?

Also, the assert library has some very basic and uninformative failure messages like "Unexpected error" and "Should be false". Please provide more context to these failures. Ideally, I shouldn't have to read the code to know the general nature of the failure.

My preference is to use standard if/else comparisons, call `testing.T`'s methods directly, and avoid assertion libraries because:

- They make it easy to omit useful messages
- They add little value over standard `if` checks.
- They often conflate comparison utilities with test failures. For example, this library has `assert.JSONEq`. It does not have `assert.JSONNotEq`. However if I had `func compareJSON(a, b string) bool`, the difference between testing equality and inequality is adding a ! in front of it. These libraries tend to end up with a cross product of comparisons and boolean operations.